cfme/tests/cloud/test_providers.py

```diff
@@ -831,4 +1007,37 @@ def test_create_azure_vm_from_azure_image():
                   VM is completely Stopped in Azure.
         title: Create Azure VM from Azure image
     """
-    pass
+    app = appliance.IPAppliance.from_url(vm_ip)
+
+    # Credentials for the provisioned VM from CFME image, this is different to the VM that runs
+    # powershell scripts as Azure has specific requirements for login/password.
+    # These credentials are used in the script create_vm.ps1 to provision the VM.
+    username = credentials['azure_appliance']['username']
+    password = credentials['azure_appliance']['password']
+
+    with ssh.SSHClient(hostname=vm_ip,
```

Comment:
I would love to see an enhancement that allows for passing username/password to `IPAppliance.__init__`, setting local vars during class construction.  Some `IPAppliance.username` and `IPAppliance.password` properties could return these local vars (`self._username`).

Then `IPAppliance.ssh_client` could use those properties, defaulting to the current conf file lookup when nothing is passed.

Then, when you create an appliance object here to test against, you can still use `app.ssh_client`, and it will automatically use the passed credentials instead of the default.

Just a thought, for a framework enhancement that would assist with this kind of testing.  There's already some code there for handling openshift creds for appliances.

Reply:
@mshriver totally agree, Mike. Although I would prefer much more to have smaller PRs and this one is growing already to editing manual tests that I didn't touch here. I can try to raise a follow-up PR to do such changes.

Reply:
oh, absolutely don't add that here, was just throwing the idea out there :)